packages/app/src/components/Sidebar/Bookmarks/BookmarkFolderItem.tsx

```diff
@@ -39,40 +46,50 @@ const BookmarkFolderItem: FC<BookmarkFolderItemProps> = (props: BookmarkFolderIt
   }, [children.length, currentChildren]);
 
   useEffect(() => {
-    if (isOpen && childBookmarkFolderData != null) {
+    if (isOpen) {
       mutateChildBookmarkData();
       setCurrentChildren(childBookmarkFolderData);
     }
   }, [childBookmarkFolderData, isOpen, mutateChildBookmarkData]);
 
+
   const hasChildren = useCallback((): boolean => {
     if (currentChildren != null && currentChildren.length > children.length) {
       return currentChildren.length > 0;
     }
     return children.length > 0;
   }, [children.length, currentChildren]);
 
-
   const loadChildFolder = useCallback(async() => {
     setIsOpen(!isOpen);
-    setCurrentParentFolder(bookmarkFolder._id);
-  }, [bookmarkFolder, isOpen]);
+    setTargetFolder(folderId);
+  }, [folderId, isOpen]);
 
+  const loadParent = useCallback(async() => {
+    mutateParentBookmarkFolder();
+  }, [mutateParentBookmarkFolder]);
 
   const onPressEnterHandler = useCallback(async(folderName: string) => {
-
     try {
-      await apiv3Post('/bookmark-folder', { name: folderName, parent: currentParentFolder });
-      setIsOpen(true);
-      setIsRenameInputShown(false);
-      mutateChildBookmarkData();
-      toastSuccess(t('Create New Bookmark Folder Success'));
+      if (isRenameAction) {
+        await apiv3Put('/bookmark-folder', { bookmarkFolderId: folderId, name: folderName, parent });
+        loadParent();
+        setIsRenameAction(false);
+        toastSuccess(t('Rename Bookmark Folder Success'));
+      }
+      else {
+        await apiv3Post('/bookmark-folder', { name: folderName, parent: targetFolder });
+        setIsOpen(true);
+        setIsRenameInputShown(false);
+        mutateChildBookmarkData();
+        toastSuccess(t('Create New Bookmark Folder Success'));
+      }
```

Comment:
Could you describe this code?
I'd like to know when this implement is executed.

Reply:
It is executed when creating a new folder or a subfolder of an existing folder (Plus button). Whereas the other lines above it are for rename . Create a new folder and rename the folder uses the same component